web/src/beta/lib/reearth-ui/components/Selector/index.tsx

```diff
@@ -0,0 +1,270 @@
+import { FC, MouseEvent, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import { styled, useTheme } from "@reearth/services/theme";
+
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+import { Popup } from "../Popup";
+import { Typography } from "../Typography";
+
+export type SelectorProps = {
+  multiple?: boolean;
+  value?: string | string[];
+  options: { value: string; label?: string }[];
+  disabled?: boolean;
+  placeholder?: string;
+  onChange?: (value: string | string[]) => void;
+};
+
+const resizeObserver = new ResizeObserver(entries => {
+  entries.forEach(entry => {
+    const selectorRef = entry.target;
+    if (selectorRef instanceof HTMLElement) {
+      const updateSelectorWidth = (selectorRef as any)._updateSelectorWidth;
+      if (updateSelectorWidth) {
+        updateSelectorWidth(selectorRef.clientWidth);
+      }
+    }
+  });
+});
+
+export const Selector: FC<SelectorProps> = ({
+  multiple,
+  value,
+  options,
+  placeholder = "Please select",
+  disabled,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const selectorRef = useRef<HTMLDivElement>(null);
+  const [selectedValue, setSelectedValue] = useState<string | string[] | undefined>(
+    value ?? (multiple ? [] : undefined),
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectorWidth, setSelectorWidth] = useState<number>();
+
+  const optionValues = useMemo(() => options, [options]);
+
+  useEffect(() => {
+    setSelectedValue(value ?? (multiple ? [] : undefined));
+  }, [value, multiple]);
+
+  const updateSelectorWidth = useCallback(() => {
+    if (selectorRef.current) {
+      setSelectorWidth(selectorRef.current.clientWidth);
+    }
+  }, []);
+
+  useEffect(() => {
+    updateSelectorWidth();
+    const handleResize = () => updateSelectorWidth();
+    return window.addEventListener("resize", handleResize);
+  }, [updateSelectorWidth]);
+
+  useEffect(() => {
+    const currentRef = selectorRef.current;
+    if (currentRef) {
+      (currentRef as any)._updateSelectorWidth = updateSelectorWidth;
+      resizeObserver.observe(currentRef);
+    }
+    return () => {
+      if (currentRef) {
+        resizeObserver.unobserve(currentRef);
+      }
+    };
+  }, [updateSelectorWidth]);
```

Comment:
```suggestion
  useEffect(() => {
    const selectorElement = selectorRef.current;
    if (!selectorElement) return;

    const resizeObserver = new ResizeObserver(entries => {
      if (!entries || entries.length === 0) return;
      const { width } = entries[0].contentRect;
      setSelectorWidth(width);
    });

    resizeObserver.observe(selectorElement);

    return () => {
      resizeObserver.unobserve(selectorElement);
      resizeObserver.disconnect();
    };
  }, []);
```
I think we can handle the resize in one single useEffect, no need to define a `updateSelectorWidth` function and no need to define a resizeObserver outside.